File: Imp_Test/resources/cpp/pizza.cpp
#include "test/Crust.h"
#include "test/Pizza.h"

namespace test {

  void Pizza::add(std::string topping) {
    auto x = 0;
    if (topping == null)
      return;
    else
      x = 1.5f;

    toppings.push_back(topping);
    crust = Crust::burnt;
  }

  std::vector<std::string> Pizza::get_eaten() {
    return toppings;
  }

}

File: Imp_Test/resources/js/pizza.js
var test = {}

test.Pizza = function() {}
test.Pizza.prototype = {
	toppings: [],
	crust: 0,
	variation: false,
	add: function(topping) {
		var x = 0
		if (topping == null)
			return
		else
			x = 1

		this.toppings.push(topping)
		this.crust = 2
	},
	get_eaten: function() {
		return this.toppings
	}
}
